### PR TITLE
CHANGE: Change extension of project-wide template asset to .json

### PR DESCRIPTION
**Pull request by jfreire-unity**
### Description

Make the project-wide actions template have a different extension https://jira.unity3d.com/browse/ISX-1588

### Changes made

Changed the extension of the file from .inputaction to .json. This way it is not imported by the InputActionImporter.
This results in not having its `InputActionReference`  appear in the object pickers. Same for the object picker of `InputActionAsset`

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Review by ekcoh — CHANGES_REQUESTED** [review]
Small change, looks good to me. However its missing a CHANGELOG entry right since this is a diff compared to pre-1?

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ however, it has to be formatted properly to pass verification tests.
 - Static analysis warnings regarding flag enums have been suppressed in order to avoid compile-time warnings or errors.
 - Action Map and Action Tree views of the UI Toolkit based Input Action Editor now prevents deselection in both views when Escape key is pressed.
 - Input Action Asset editors Auto-save feature has been modified to trigger on focus-lost when activated instead of triggering on every modification to the asset in order to reduce impact of processing required to handle modified assets.
+- Project-wide input actions template extension changed from .inputactions to .json.
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Why would a user care about this? It's an internal detail, they don't use the template.

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ namespace UnityEngine.InputSystem.Editor
 {
     internal static class ProjectWideActionsAsset
     {
-        internal const string kDefaultAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.inputactions";
+        internal const string kDefaultAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActions.json";
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs]: No longer a template?

**Review by jamesmcgill — APPROVED** [review]
Looks good.

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (lyndon-unity) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: True but we could add something to say "so it no longer shows up in the action's selector UI as its not intended to be used directly, as its just a template for the generated project wide input actions in the project settings"

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs (hunk 2 above)
Comment (lyndon-unity) [reply Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs]: Agreed - Would suggest "Template" should still be in the name to make it clear

Comment (lyndon-unity) [reply Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs]: (File still contains this in it with the Template in the name : "name": "ProjectWideActionsTemplate" so keeping name to that with .json appended would make sense)

**Review by lyndon-unity — CHANGES_REQUESTED** [review]
Minor suggest to keep "Template" in the name and expand slightly on change log but overall I think the change is good to hide it from the actions UI areas.

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs (hunk 2 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs]: Changed 👍

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Changed

**Review by lyndon-unity — APPROVED** [review]

**Comment by lyndon-unity** [comment]
Tested locally to confirm the template actions don't show up :)

**Review by ekcoh — APPROVED** [review]
LGTM

**Pull request merged**